src/HwmonTempSensor.cpp

```diff
@@ -56,13 +55,21 @@ HwmonTempSensor::HwmonTempSensor(
            std::move(thresholdsIn), sensorConfiguration, objectType, false,
            false, thisSensorParameters.maxValue, thisSensorParameters.minValue,
            conn, powerState),
-    i2cDevice(i2cDevice), objServer(objectServer),
-    inputDev(io, path, boost::asio::random_access_file::read_only),
-    waitTimer(io), path(path), offsetValue(thisSensorParameters.offsetValue),
+    i2cDevice(i2cDevice), objServer(objectServer), inputDev(io), waitTimer(io),
+    path(path), offsetValue(thisSensorParameters.offsetValue),
     scaleValue(thisSensorParameters.scaleValue),
     sensorPollMs(static_cast<unsigned int>(pollRate * 1000)), bus(bus),
     address(address)
 {
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg)
+    int fd = open(path.c_str(), O_RDONLY);
+    if (fd < 0)
+    {
+        std::cerr << "HwmonTempSensor " << sensorName << " failed to open "
+                  << path << "\n";
+    }
+    inputDev.assign(fd);
+
```

Comment:
Just curious, I noticed in the activate() method it calls markAvailable() and setupRead() after setting up inputDev.  Is that not necessary here?